### PR TITLE
Row selection focus

### PR DESCRIPTION
**Pull request by cm9361**
### Summary

**What was changed:**
The WorklistDataGrid was updated such that the initial focus is set to the row selection column header instead of the first column selection cell.

**Why it was changed:**
The change was made after integration testing found that the requirement update was necessary for a better user experience.

### Testing

This change was tested using:

- [x] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [x] Functional review

### Additional Details
N/A

**This PR resolves:**

UXPLATFORM-9612 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Review by kenk2 — APPROVED** [review]
One small nit, otherwise LGTM!

File: packages/terra-worklist-data-grid/tests/jest/WorklistDataGrid.test.jsx [hunk 1]
```diff
@@ -491,6 +491,9 @@ describe('Row selection', () => {
       />,
     );
 
+    const rowSelectionHeader = wrapper.find(ColumnHeaderCell).at(0);
+    rowSelectionHeader.at(0).simulate('keydown', { keyCode: 40 });
```
[the remark this answers is not in the document]
Comment (cm9361) [reply packages/terra-worklist-data-grid/tests/jest/WorklistDataGrid.test.jsx:495]: I would rather not impact other test scenarios and cause potential merge conflicts at this time.  However, I do think this is something we could do as cleanup, if this is the agreed upon change.  Currently, this logic is consistent with the rest of the file.

[the remark this answers is not in the document]
Comment (kenk2) [reply packages/terra-worklist-data-grid/tests/jest/WorklistDataGrid.test.jsx]: @cm9361 Sounds good!

File: packages/terra-data-grid/src/WorklistDataGrid.jsx [hunk 2]
```diff
@@ -269,31 +270,28 @@ function WorklistDataGrid(props) {
 
   // useEffect for row selection
   useEffect(() => {
+    if (!rowSelectionEffectTriggered.current) {
+      rowSelectionEffectTriggered.current = true;
+      return;
+    }
+
```
Comment (KoushikKalli) [inline packages/terra-data-grid/src/WorklistDataGrid.jsx:277]: When the consumer loads the data grid with row selection turned on for the first time, should we announce that row selection is turned on?
The requirement does not specify this scenario and we don't think this issue is isolated this JIRA or row selection. The question applies if the consumer loads with some rows selected or cells selected. Do we expect to announce messages for each one of them? @eawww, thoughts?

**Review by KoushikKalli — APPROVED** [review]

**Pull request merged**